metrics/ci_integrations/test/destination/firestore/adapter/firestore_destination_client_adapter_test.dart

```diff
@@ -79,20 +79,21 @@ void main() {
     );
 
     test(
-      ".addBuilds() returns normally if firestore throws GrpcError.notFound",
+      ".addBuilds() throws an exception if the firestore throws exception different from GrpcError.notFound",
       () {
-        whenFetchProject().thenThrow(GrpcError.notFound());
+        whenFetchProject().thenThrow(Exception());
 
         final result = adapter.addBuilds(testProjectId, []);
 
-        expect(result, completes);
+        expect(result, throwsException);
       },
     );
 
     test(
-      ".addBuilds() throws an exception if the firestore throws exception different from GrpcError.notFound",
+      ".addBuilds() throws an exception if a project with the given id does not exist",
```

Comment:
```suggestion
      ".addBuilds() throws an ArgumentError if a project with the given id does not exist",
```
And please rewrite the test to match the behavior.